src/app/search/state/search.service.ts

```diff
@@ -166,6 +169,13 @@ export class SearchService {
 
   @transaction()
   setSearchText(text: string) {
+    if (text.length > searchTermLengthLimit) {
+      text = '';
+      this.alertService.customDetailedError(
```

Comment:
Same question as other comment -- are you displaying this before the request is even sent?

Also:

1. It would be good to avoid the duplication, but it might be hard to find a good place to put it.
2. I'm wary of a function behaving unexpectedly like this without any indication that it ignored the input (the error message is for the end user, not the code that invoked this method). But this is an extreme edge case presumably, most likely only run into somebody trying to break the code.

Reply:
Yeah, I'm displaying before ever sending a request so I'll remove the status code. 

And I'm going to remove one of these alerts because it wasn't actually doing anything... If a user performs a search using the search box, they'll get the red alert box. If they do a search by using a url, then the search term gets replaced with the empty string, which causes the permalink to be updated. There won't be an error displayed to the user, the page just loads like navigating to /search. But yeah I can't imagine a user using such a long term for a genuine search so I think this is okay.